sdk/formrecognizer/Azure.AI.FormRecognizer/src/CopyAuthorization.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+
+namespace Azure.AI.FormRecognizer.Training
+{
+    /// <summary>
+    /// Authorization for copying a custom model into the target Form Recognizer resource.
+    /// </summary>
+    public class CopyAuthorization
+    {
+        /// <summary>Model identifier in the target Form Recognizer Resource. </summary>
+        public string ModelId { get; }
+        /// <summary> The time when the access token expires. The date is represented as the number of seconds from 1970-01-01T0:0:0Z UTC until the expiration time. </summary>
+        //public DateTimeOffset ExpiresOn { get; set; }
+        public long ExpiresOn { get; }
+        /// <summary> Token claim used to authorize the request. </summary>
+        internal string _accessToken { get; }
+        /// <summary> Azure Resource Id of the target Form Recognizer resource where the model is copied to. </summary>
+        internal string _resourceId { get; }
+        /// <summary> Location of the target Form Recognizer resource. A valid Azure region name supported by Cognitive Services. </summary>
+        internal string _region { get; }
+
+        internal CopyAuthorization(string modelId, string accessToken, long expirationDateTimeTicks, string resourceId, string region)
+        {
+            ModelId = modelId;
+            _accessToken = accessToken;
+            //ExpiresOn = DateTimeOffset.FromUnixTimeSeconds(expirationDateTimeTicks);
+            ExpiresOn = expirationDateTimeTicks;
+            _resourceId = resourceId;
+            _region = region;
+        }
+
+        internal CopyAuthorization(CopyAuthorizationResult copyAuth, string resourceId, string region)
+            : this(copyAuth.ModelId, copyAuth.AccessToken, copyAuth.ExpirationDateTimeTicks, resourceId, region) { }
+
+        /// <summary>
+        /// Deserializes an opaque string into a <see cref="CopyAuthorization"/> object.
+        /// </summary>
+        /// <param name="accessToken">Opaque string with the access token information for a specific model.</param>
+        public static CopyAuthorization FromJson(string accessToken)
+        {
+            CopyAuthorizationParse parse = JsonSerializer.Deserialize<CopyAuthorizationParse>(accessToken);
+            return new CopyAuthorization(
```

Comment:
property names don't match to what the values of the json are, so I need that intermediate class.
If there is an easier way to do it, happy to change it

Reply:
It looks like you may be able to do this with the [`JsonPropertyName`](https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-how-to#customize-individual-property-names) attribute

Reply:
I looked into it and currently, it only supports public properties. For private or internal we need to do a custom converter => https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to#public-and-non-public-fields

How strongly do you feel about going that route instead of the private class I created

Reply:
If this were a high-traffic path, I would be concerned about a possible perf issue, but since we don't expect this API to be used very often, I'm fine with it this way.